### PR TITLE
Add linting with black and isort

### PR DESCRIPTION
**Pull request by robhudson**
Without the commits to reformat the code using `black` and `isort`, this will break the build. Those changes will come in follow up commits. But first I wanted to test and make sure these checks will fail before committing the correcting changes.

Any comments on execution here is welcomed.

### CODE REVIEW

**Comment by robhudson** [comment]
We may want to add some docs in the developer docs about conforming to the isort and black standards? Are there any other things we need to do here?

**Comment by bcolloran** [comment]
> We may want to add some docs in the developer docs about conforming to the isort and black standards? 

Yeah, that would be great! I don't know of anything else to add, perhaps @wlach or @hamilton have ideas tho

File: Makefile [hunk 1]
```diff
@@ -14,8 +14,11 @@ shell:
 up:
 	docker-compose up
 
-flake8:
-	docker-compose run server flake8 server/
+lint:
```
Comment (wlach) [inline Makefile]: These new targets should be marked as `.PHONY` at the top of the file, as they don't depend on any files changing.

**Comment by wlach** [comment]
Looks great, I will merge as soon as I confirm CI is passing

**Comment by wlach** [comment]
Thanks so much!

**Pull request merged**